### PR TITLE
OK #2642 - Update the invitation flow for public events

### PR DESCRIPTION
**Pull request by phpgeekfr**

### CODE REVIEW

File: apps/backend-functions/src/internals/invitations/events.ts [hunk 1]
```diff
@@ -20,44 +21,53 @@ async function onInvitationToAnEventCreate({
...
 
-  let recipient: string;
-  if (!!toUser) {
-    recipient = toUser.email;
-  } else if (!!toOrg) {
-    /** Attendee is only an user or an email for now */
-    throw new Error('Cannot invite an org to an event for now. Not implemented.');
+  // fetch event
+  const eventSnapshot = await db.doc(`events/${eventId}`).get();
+  const event = eventSnapshot.data();
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/internals/invitations/events.ts:26]: Don't we have something like 
```typescript
const event = await getSnapshot(`events/${eventId}`)
```

File: apps/backend-functions/src/internals/invitations/events.ts [hunk 2]
```diff
@@ -20,44 +21,53 @@ async function onInvitationToAnEventCreate({
...
+
+  if (mode === 'request' && event?.isPrivate === false) {
+    // This will then trigger "onInvitationToAnEventAccepted" and send in-app notification to 'fromUser' 
+    await db.doc(`invitations/${id}`).set({ status: 'accepted' }, { merge: true });
   } else {
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/internals/invitations/events.ts:31]: Why you have a `else` here ?

File: apps/backend-functions/src/internals/invitations/events.ts [hunk 3]
```diff
@@ -20,44 +21,53 @@ async function onInvitationToAnEventCreate({
...
-    throw new Error('Who is this invitation for ?');
-  }
+    let recipient: string;
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/internals/invitations/events.ts:32]: This is weird for me to scrope the recippient in the else. 
What I mean is that it's difficult to understand what's doing what with all the if/else

File: apps/backend-functions/src/internals/invitations/events.ts [hunk 4]
```diff
@@ -20,44 +21,53 @@ async function onInvitationToAnEventCreate({
...
+    if (!!toUser) {
+      recipient = toUser.email;
+    } else if (!!toOrg) {
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/internals/invitations/events.ts:35]: Can you do all the protective if statement at the beginning ?

File: libs/notification/src/lib/invitation/invitation-item/invitation-item.component.ts [hunk 5]
```diff
@@ -28,7 +26,25 @@ export class InvitationItemComponent {
       case 'fromOrganizationToUser':
         return `Your organization sent an invitation to this user email: ${this.invitation.toUser.email}`;
       case 'event':
-        return this.service.isInvitationForMe(this.invitation) ? `You have been invited to an event !` : `Your invitation have been sent!`;
+        return this.getMessageToDisplay();
+    }
+  }
+
+  private getMessageToDisplay() {
```
Comment (GrandSchtroumpf) [inline libs/notification/src/lib/invitation/invitation-item/invitation-item.component.ts:33]: There is another 'invitation-item' component inside the invitations/component/item folder. This is the one that we use in festival & will keep for the future.
I think it's better to have this code somewhere else. For me the best would be to have a pure function & call it in `formatFromFirestore` : 
```typescript
function getMessage(invitation, isForMe) {
  const { mode, fromOrg, fromUser} = invitation;
  if (isForMe) {
    if (mode === 'request')
    if (fromOrg) {
      const { pulic, full } = fromOrg.denomination;
      from = public || full;
    } else if (fromUser) {
      const { firstName, lastName, email } = fromUser;
      from = firstName && lastName ? ${} : email;
    }
    ....
  }
  ....
}
```
And in formatFromFirestore : 
```typescript
formatFromFirestore(invitation) {
   ...
   const isForMe = this.isInvitationForMe(invitation);
   const msg = getMessage(invitation, isForMe);
   ....
}
```

File: libs/organization/src/lib/organization/components/member-request/member-request.component.ts [hunk 6]
```diff
@@ -22,18 +22,18 @@ export class MemberRequestComponent {
...
     return this.isAdmin
-    ? [ 'name', 'surname', 'email', 'uid' ]
-    : [ 'name', 'surname', 'email' ]
+    ? [ 'firstName', 'lastName', 'email', 'uid' ]
+    : [ 'firstName', 'lastName', 'email' ]
   }
 
   public findInvitation(uid: string) {
-    return this.memberInvitations.find(invitation => invitation.fromUser.uid === uid);
+    return this.memberInvitations.find(invitation => (invitation.fromUser.uid === uid && invitation.type === 'fromUserToOrganization'));
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/organization/components/member-request/member-request.component.ts:37]: We should change it after my PR.

File: libs/organization/src/lib/organization/pages/member/member.component.ts [hunk 7]
```diff
@@ -49,11 +49,11 @@ export class MemberComponent implements OnInit {
     this.isSuperAdmin$ = this.permissionQuery.isSuperAdmin$;
 
     this.invitationsFromOrganization$ = this.db
-      .collection('invitations', ref => ref.where('fromOrg.id', '==', this.query.getActiveId()).where('status', '==', 'pending'))
+      .collection('invitations', ref => ref.where('fromOrg.id', '==', this.query.getActiveId()).where('status', '==', 'pending').where('fromOrg.id', '==', this.query.getActiveId()).where('type', '==', 'fromOrganizationToUser'))
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/organization/pages/member/member.component.ts:52]: type should be change here after my PR.

File: libs/organization/src/lib/organization/pages/member/member.component.ts [hunk 8]
```diff
@@ -49,11 +49,11 @@ export class MemberComponent implements OnInit {
...
       .valueChanges() as Observable<Invitation[]>;
 
     this.invitationsToJoinOrganization$ = this.db
-      .collection('invitations', ref => ref.where('toOrg.id', '==', this.query.getActiveId()).where('status', '==', 'pending'))
+      .collection('invitations', ref => ref.where('toOrg.id', '==', this.query.getActiveId()).where('status', '==', 'pending').where('type', '==', 'fromUserToOrganization'))
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/organization/pages/member/member.component.ts:56]: Don't repeat yourself with `getActiveId()`, create a variable
```typescript
const id = this.query.getActiveId();
```

File: libs/organization/src/lib/organization/pages/member/member.component.ts (hunk 7 above)
Comment (GrandSchtroumpf) [reply libs/organization/src/lib/organization/pages/member/member.component.ts:52]: Also you wrote twice `.where('fromOrg.id', '==', this.query.getActiveId())`

File: libs/organization/src/lib/organization/pages/member/member.component.ts [hunk 9]
```diff
@@ -49,11 +49,11 @@ export class MemberComponent implements OnInit {
     this.isSuperAdmin$ = this.permissionQuery.isSuperAdmin$;
 
     this.invitationsFromOrganization$ = this.db
-      .collection('invitations', ref => ref.where('fromOrg.id', '==', this.query.getActiveId()).where('status', '==', 'pending'))
+      .collection('invitations', ref => ref.where('fromOrg.id', '==', this.query.getActiveId()).where('status', '==', 'pending').where('fromOrg.id', '==', this.query.getActiveId()).where('type', '==', 'fromOrganizationToUser'))
       .valueChanges() as Observable<Invitation[]>;
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/organization/pages/member/member.component.ts:53]: Change for :
```typescript
const queryFn = ref => ref.where('fromOrg.id', '==', id).where('status', '==', 'pending').where('type', '==', '');
this.invitationsFromOrganization$ = this.invitationService.valueChanges(queryFn),
```
Like that you won't have to force the type.

File: apps/backend-functions/src/internals/invitations/events.ts (hunk 2 above)
Comment (phpgeekfr) [reply apps/backend-functions/src/internals/invitations/events.ts:31]: cuz, il invitation is a request to a public event, we do not need to execute the code after else.
But I will do a return instead

File: apps/backend-functions/src/internals/invitations/events.ts (hunk 1 above)
Comment (phpgeekfr) [reply apps/backend-functions/src/internals/invitations/events.ts:26]: no :)

File: apps/backend-functions/src/internals/invitations/events.ts (hunk 4 above)
Comment (phpgeekfr) [reply apps/backend-functions/src/internals/invitations/events.ts:35]: If I move it above, we will have some cases where the code is executed but not used

Comment (GrandSchtroumpf) [reply apps/backend-functions/src/internals/invitations/events.ts:35]: I'm not sure to follow. If you throw, better throw at the beginning than in the middle of the function.

**Pull request merged**